field-agent/src/main/java/learn/field_agent/data/AliasJdbcTemplateRepository.java

```diff
@@ -0,0 +1,100 @@
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+import java.sql.PreparedStatement;
+import java.sql.Statement;
+import java.util.List;
+
+@Repository
+public class AliasJdbcTemplateRepository implements AliasRepository{
+    private final JdbcTemplate jdbcTemplate;
+
+    public AliasJdbcTemplateRepository(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    @Override
+    public List<Alias> findAll() {
+        final String sql = "select alias_id, name, persona, agent_id "
+                + "from alias ";
+
+        return jdbcTemplate.query(sql, new AliasMapper());
+    }
+
+    @Override
+    public Alias findById(int aliasId) {
+        final String sql = "select alias_id, name, persona, agent_id "
+                + "from alias "
+                + "where alias_id = ?;";
+
+        return jdbcTemplate.query(sql, new AliasMapper(), aliasId).stream()
+                .findFirst()
+                .orElse(null);
+    }
+
+    @Override
+    public List<AgentAlias> findByAlias(String aliasName) {
+        final String sql = "select ag.agent_id, ag.first_name, ag.middle_name, ag.last_name, ag.dob, ag.height_in_inches, als.alias_id, als.name, als.persona "
+        + "from agent ag "
+        + "inner join alias als "
+        + "on als.agent_id = ag.agent_id "
+        + "where als.name=?;";
+
+        return jdbcTemplate.query(sql, new AgentAliasMapper(), aliasName);
+    }
+
+    @Override
+    public Alias add(Alias alias) {
+        final String sql = "insert into alias (name, persona, agent_id)"
+                + "values (?,?,?);";
+
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+        int rowsAffected = jdbcTemplate.update(connection -> {
+            PreparedStatement ps = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);
+            ps.setString(1, alias.getName());
+            ps.setString(2, alias.getPersona());
+            ps.setInt(3, alias.getAgentId());
+            return ps;
+        }, keyHolder);
+
+        if (rowsAffected <= 0) {
+            return null;
+        }
+
+        alias.setAliasId(keyHolder.getKey().intValue());
+        return alias;
+    }
+
+    @Override
+    public boolean update(Alias alias) {
+        final String sql = "update alias set "
+                + "name = ?, "
+                + "persona = ?, "
+                + "agent_id = ? "
+                + "where alias_id = ? and agent_id = ?;";
+
+
+        return jdbcTemplate.update(sql,
+                alias.getName(),
+                alias.getPersona(),
+                alias.getAgentId(),
+                alias.getAliasId(),
+                alias.getAgentId()) > 0;
+    }
+
+    @Override
+    public boolean deleteById(int aliasId) {
+        return jdbcTemplate.update(
+                "delete from alias where alias_id = ?", aliasId) > 0;
+    }
+}
```

Comment:
nice job on implementing all of the new concepts that you learned e.g. jdbcTemplate, mappers, etc...